src/edu/csus/ecs/pc2/clics/API202306/AwardService.java

```diff
@@ -0,0 +1,260 @@
+            }
+        }
+        return Response.status(Response.Status.NOT_FOUND).build();
+    }
+
+    /**
+     * Post a new award.
+     * 
+     * @param servletRequest details of request
+     * @param sc requesting user's authorization info
+     * @param contestId The contest
+     * @param jsonInputString citation and team_ids json for the new award
+     * @return json for the award, including the id
+     */
+    @POST
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    @Path("{contestId}/awards")
+    public Response addNewAward(@Context HttpServletRequest servletRequest, @Context SecurityContext sc, @PathParam("contestId") String contestId, String jsonInputString) {
+        
+        if(sc.isUserInRole(WebServer.WEBAPI_ROLE_ADMIN) == false) {
+            return Response.status(Response.Status.FORBIDDEN).build();
+        }
+        
+        return Response.status(Response.Status.NOT_IMPLEMENTED).build();
+    }
+    
+    /**
+     * Put updates an existing award
+     * 
+     * @param servletRequest details of request
+     * @param sc requesting user's authorization info
+     * @param contestId The contest
+     * @param jsonInputString citation and team_ids json for the new award
+     * @return json for the award, including the id
+     */
+    @PUT
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    @Path("{contestId}/awards/{awardId}")
+    public Response setAwardInfo(@Context HttpServletRequest servletRequest, @Context SecurityContext sc, @PathParam("contestId") String contestId, @PathParam("awardId") String awardId, String jsonInputString) {
+        
+        if(sc.isUserInRole(WebServer.WEBAPI_ROLE_ADMIN) == false) {
+            return Response.status(Response.Status.FORBIDDEN).build();
+        }
+        
+        return Response.status(Response.Status.NOT_IMPLEMENTED).build();
+    }
+    
+    /**
+     * Patch updates an existing award
+     * 
+     * @param servletRequest details of request
+     * @param sc requesting user's authorization info
+     * @param contestId The contest
+     * @param jsonInputString citation and team_ids json for the new award
+     * @return json for the award, including the id
+     */
+    @PATCH
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    @Path("{contestId}/awards/{awardId}")
+    public Response updateAwardProperty(@Context HttpServletRequest servletRequest, @Context SecurityContext sc, @PathParam("contestId") String contestId, @PathParam("awardId") String awardId, String jsonInputString) {
+        
+        if(sc.isUserInRole(WebServer.WEBAPI_ROLE_ADMIN) == false) {
+            return Response.status(Response.Status.FORBIDDEN).build();
+        }
+        
+        return Response.status(Response.Status.NOT_IMPLEMENTED).build();
```

Comment:
It seems like this condition should be logged...

Reply:
See above.